### ISSUE TITLE
Firestorm-bin updated URL for the linux package and now results in a 404 error

### ISSUE THREAD

**Issue by zewm** [lmiphay/gentoo.overlay#18]
When trying to emerge this package (games-simulation/Firestorm-bin-6.5.3-r1) it results in a 404 error when attempting to download the linux package/source from the firestorm website.

In the current ebuild it points to 
     https://downloads.firestormviewer.org/preview/linux/Phoenix_Firestorm-Releasex64_x86_64_6.5.3.65658.tar.xz

but the URL has changed (/preview/ has been omitted) and is now this url
     https://downloads.firestormviewer.org/linux/Phoenix_Firestorm-Releasex64_x86_64_6.5.3.65658.tar.xz

The version numbers are all correct and have not changed as of this writing.

source: https://www.firestormviewer.org/linux/

**Comment by lmiphay** [comment]
Thanks for the report @zewm - updated now.

**Comment by lmiphay** [comment]
I think that firestorm and/or cloudfare have recently restricted downloads to those using web browsers only.

This would be fetch restricted in gentoo terms.

The workaround is to download the tarball using the web browser and manually move it to /usr/portage/disfiles

I will update the ebuild with instructions to do this, if it turns out that the situation is permanent...

**Comment by ntheo1979** [comment]
It seems you are right, it happens with release versions too. I do not have a /usr/portage/distfiles (I do not have a /usr/portage at all), should I make one?

**Comment by lmiphay** [comment]
You may have a /var/cache/distfiles ? 

If you do have that directory, then that is where the tarball should be moved after downloading it with a web browser.

Some notes on handling fetch restricted packages:

1. https://www.seifertm.de/blog/fetching-packages-from-authenticated-http-uris-with-gentoo-portage/
2. https://devmanual.gentoo.org/ebuild-writing/functions/pkg_nofetch/index.html

**Comment by ntheo1979** [comment]
Well, that did it. Thank you, and for the references too.

**Comment by lmiphay** [comment]
The  latest release candidate downloaded directly via the ebuild.

Let me know if you see further fetch failures. I will add a message anyway in case the DDoS protection is turned on again on the fs website.

**Comment by ntheo1979** [comment]
Yes, I see this: https://jira.firestormviewer.org/browse/FIRE-33379 Did you file it?

Thanks for working on it :) Firestorm can be tiresome to work sometimes, but alas it is the viewer with the most features.
